modules/lang-painless/src/test/java/org/elasticsearch/painless/GetByPathAugmentationTests.java

```diff
@@ -0,0 +1,183 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class GetByPathAugmentationTests extends ScriptTestCase {
+
+    private final String k001Key = "k011";
+    private final String k001Value = "b";
+    private final Map<String,String> k001Obj = new HashMap<>();
+    private final String k001MapStr = "['" + k001Key + "': '" + k001Value + "']";
+    private final String mapMapList = "['k0': ['k01': [['k010': 'a'], " + k001MapStr + "]], 'k1': ['q']]";
+
+    private final String l2m2l1Index0 = "ll0";
+    private final String l2m2l1Index1 = "ll1";
+    private final List<String> l2m2l1Obj = new ArrayList<>();
+    private final String l2m2l1Str = "['" + l2m2l1Index0 + "', '" + l2m2l1Index1 + "']";
+    private final String listMapListList = "[['m0':'v0'],['m1':'v1'],['m2':['l0','l1', " + l2m2l1Str + "]]]";
+
+    public GetByPathAugmentationTests() {
+        l2m2l1Obj.add(l2m2l1Index0);
+        l2m2l1Obj.add(l2m2l1Index1);
+        k001Obj.put(k001Key, k001Value);
+    }
+
+    private String toScript(String collection, String key) {
+        return String.format(Locale.ROOT, "return %s.getByPath('%s')", collection, key);
+    }
+
+    private String toScript(String collection, String key, String defaultValue) {
+        return String.format(Locale.ROOT, "return %s.getByPath('%s', %s)", collection, key, defaultValue);
+    }
+
+    private IllegalArgumentException numberFormat(String unparsable, String path, int i) {
+        String format = "Could not parse [%s] as a int index into list at path [%s] and index [%d]";
+        return new IllegalArgumentException(String.format(Locale.ROOT, format, unparsable, path, i,
+            new NumberFormatException()));
+    }
+
+    private IllegalArgumentException missingValue(String path) {
+        String format = "Could not find value at path [%s]";
+        return new IllegalArgumentException(String.format(Locale.ROOT, format, path));
+    }
+
+
+    private void assertPathValue(String collection, String key, Object value) {
+        assertEquals(value, exec(toScript(collection, key)));
+    }
+
+    private void assertPathDefaultValue(String collection, String key, Object value, String defaultValue) {
+        assertEquals(value, exec(toScript(collection, key, defaultValue)));
+    }
+
+    private void assertPathError(String collection, String key, IllegalArgumentException error) {
+        assertPathError(toScript(collection, key), error);
+    }
+
+    private void assertPathError(String collection, String key, String defaultValue, IllegalArgumentException error) {
+        assertPathError(toScript(collection, key, defaultValue), error);
+    }
+
+    private void assertPathError(String script, IllegalArgumentException error) {
```

Comment:
We don't need to reproduce the exception to check it. Just pass in the message here that we want to check for. You can even just check for a substring with the `containsString` matcher.

Reply:
Just passing in the message now.